Source/WebCore/platform/graphics/gstreamer/VideoEncoderGStreamer.cpp

```diff
@@ -165,14 +164,53 @@ void GStreamerVideoEncoder::close()
     m_internalEncoder->close();
 }
 
-GStreamerInternalVideoEncoder::GStreamerInternalVideoEncoder(const String& codecName, VideoEncoder::OutputCallback&& outputCallback, VideoEncoder::PostTaskCallback&& postTaskCallback)
+GStreamerInternalVideoEncoder::GStreamerInternalVideoEncoder(const String& codecName, VideoEncoder::DescriptionCallback&& descriptionCallback, VideoEncoder::OutputCallback&& outputCallback, VideoEncoder::PostTaskCallback&& postTaskCallback)
     : m_codecName(codecName)
+    , m_descriptionCallback(WTFMove(descriptionCallback))
     , m_outputCallback(WTFMove(outputCallback))
     , m_postTaskCallback(WTFMove(postTaskCallback))
 {
     GRefPtr<GstElement> element = gst_element_factory_make("webkitvideoencoder", nullptr);
-    m_harness = GStreamerElementHarness::create(WTFMove(element), [protectedThis = Ref { *this }, this](auto&, const GRefPtr<GstBuffer>& outputBuffer) {
-        if (protectedThis->m_isClosed)
+
+    auto pad = adoptGRef(gst_element_get_static_pad(element.get(), "src"));
+    g_signal_connect(pad.get(), "notify::caps", G_CALLBACK(+[](GObject* pad, GParamSpec*, gpointer userData) {
+        auto* encoder = static_cast<WeakPtr<GStreamerInternalVideoEncoder>*>(userData);
+        if (!encoder)
+            return;
```

Comment:
As I mentioned in the other review, this is not doing what you'd expect.

Something like:

`WeakPtr<GStreamerInternalVideoEncoder> encoder(*static_cast<WeakPtr<GStreamerInternalVideoEncoder>*>(userData));`

Then you'd need to use g_signal_connect_data to provide the GDestroyNotify that will just delete the WeakPtr.

Reply:
Done, and added a `g_signal_handlers_disconnect_by_data()` too.